src/caches.jl

```diff
@@ -113,14 +123,19 @@ function OrdinaryDiffEq.alg_cache(
 
     # Measurement model related things
     R = zeros(uElType, d, d)
-    H = f isa DynamicalODEFunction ? copy(E2) : copy(E1)
+    H_base = if f isa DynamicalODEFunction
```

Comment:
Not yet sure if this is the correct place for this logic. Maybe rather somehow dispatch on the measurement_model type.

Also, if the mass_matrix were to be changed during the solve, we would not get correct results.